money/money.py

```diff
@@ -1,16 +1,39 @@
 """Class representing a monetary amount"""
 
 from typing import Union
-from decimal import Decimal, ROUND_HALF_UP
+from decimal import Decimal
+from decimal import ROUND_CEILING
+from decimal import ROUND_DOWN
+from decimal import ROUND_FLOOR
+from decimal import ROUND_HALF_DOWN
+from decimal import ROUND_HALF_EVEN
+from decimal import ROUND_HALF_UP
+from decimal import ROUND_UP
+from decimal import ROUND_05UP
 from babel.numbers import format_currency
 from money.currency import Currency
 from money.currency import CurrencyHelper
 from money.exceptions import InvalidAmountError, CurrencyMismatchError, InvalidOperandError
 
+# pylint: disable=invalid-name
+DecimalRoundingType = Union[
+    ROUND_CEILING,
+    ROUND_DOWN,
+    ROUND_FLOOR,
+    ROUND_HALF_DOWN,
+    ROUND_HALF_EVEN,
+    ROUND_HALF_UP,
+    ROUND_UP,
+    ROUND_05UP,
+]
+
 class Money:
     """Class representing a monetary amount"""
 
-    def __init__(self, amount: str, currency: Currency=Currency.USD) -> None:
+    rounding_per_operation = True
```

Comment:
Is rounding_per_operation just a way to enable or disable rounding? if rounding is disabled, you could end up with a monetary amount that is invalid, for example, USD 3.105 (invalid, since USD only supports 2 decimal places). In this case, the Money constructor will throw an error. 

Although fractional amounts like that do have some applications, they are not needed in most applications and open up a pretty big can of worms. As the README for this library says:

> If you're doing complicated money operations that require many digits of precision for some reason (or you're running a gas station and charging that extra nine tenths of a cent), this library is not for you.

I am open to adding support for that type of arithmetic to this library, but it will require readme updates, conditionally disabling the code that currently forbids it, and thorough unit testing. One other idea: a ROUND_NONE DecimalRoundingType may be more clear than a separate rounding_per_operation flag. If you would like to add the ability to disable rounding, that may be better off as a separate Pull Request.